app/components/post_list/post_list.js

```diff
@@ -24,37 +24,74 @@ const style = StyleSheet.create({
     }
 });
 
-export default class PostList extends React.Component {
+const LOAD_MORE_POSTS = 'load-more-posts';
+
+export default class PostList extends Component {
     static propTypes = {
-        posts: React.PropTypes.array.isRequired,
-        theme: React.PropTypes.object.isRequired,
-        onPostPress: React.PropTypes.func,
-        renderReplies: React.PropTypes.bool,
-        indicateNewMessages: React.PropTypes.bool,
-        currentUserId: React.PropTypes.string,
-        lastViewedAt: React.PropTypes.number
+        posts: PropTypes.array.isRequired,
+        postsRequests: PropTypes.object.isRequired,
+        theme: PropTypes.object.isRequired,
+        allowLoadMore: PropTypes.bool,
+        loadMore: PropTypes.func,
+        onPostPress: PropTypes.func,
+        renderReplies: PropTypes.bool,
+        indicateNewMessages: PropTypes.bool,
+        currentUserId: PropTypes.string,
+        lastViewedAt: PropTypes.number
     };
 
     constructor(props) {
         super(props);
-        const {posts, indicateNewMessages, currentUserId, lastViewedAt} = this.props;
         this.state = {
+            hasFirstPost: false,
+            didInitialPostsLoad: false,
             dataSource: new ListView.DataSource({
                 rowHasChanged: (a, b) => a !== b
-            }).cloneWithRows(addDatesToPostList(posts, indicateNewMessages, currentUserId, lastViewedAt))
+            }).cloneWithRows(this.getPostsWithDates(props))
         };
     }
 
     componentWillReceiveProps(nextProps) {
+        const didInitialPostsLoad = this.didPostsLoad(nextProps, 'getPosts') ||
```

Comment:
This whole logic for figuring out if we've done the initial load or anything should be in the `ChannelPostList`, not the `PostList` itself. Other places where we'll be using a `PostList` like the thread view or search won't care about all this since their data will already be loaded or will be loaded via different requests.

Ideally, I think it should be that `ChannelPostList` looks at the store state and uses that to determine if we should pass a `loadMore` prop to the `PostList` based on whether or not it has the first post (which would be nice if it could be saved in the store). `ChannelPostList` would also pass in whether or not more posts were loading as well so that the `PostList` wouldn't have to care at all what was going on to load those posts